src/connectors/NativeConnectorLogic.sol

```diff
@@ -12,50 +12,35 @@ abstract contract NativeConnectorLogic is TokenConnectorLogic {
     event Locked(address indexed account, uint256 value);
 
     /**
-     * @dev Applies the given state transferring TARA to the specified accounts
+     * @dev Applies the given state to the token contract by transfers.
      * @param _state The state to be applied.
-     * @return accounts Affected accounts that we should split fee between
      */
-    function applyState(bytes calldata _state) internal override returns (address[] memory accounts) {
+    function applyState(bytes calldata _state) public virtual override {
         Transfer[] memory transfers = deserializeTransfers(_state);
-        accounts = new address[](transfers.length);
         uint256 transfersLength = transfers.length;
-        for (uint256 i = 0; i < transfersLength; i++) {
-            toClaim[transfers[i].account] += transfers[i].amount;
-            accounts[i] = transfers[i].account;
-            emit ClaimAccrued(transfers[i].account, transfers[i].amount);
+        for (uint256 i = 0; i < transfersLength;) {
+            (bool success,) = payable(transfers[i].account).call{value: transfers[i].amount}("");
+            if (!success) {
+                revert TransferFailed(transfers[i].account, transfers[i].amount);
+            }
+            unchecked {
+                ++i;
+            }
         }
     }
-
+    
     /**
      * @dev Locks the specified amount of tokens to transfer them to the other network.
      * @notice This function is payable, meaning it can receive TARA.
+     * @param value The amount of tokens to lock.
      */
-    function lock() public payable {
-        if (msg.value == 0) {
+    function lock(uint256 value) public payable onlySettled(value, true) {
```

Comment:
Why do you need this value if you can use msg.value here? 
Looks weird that you even not locking this exact one, but calculating from the `msg.value - settlementFee`

Reply:
because you need it in the other connectors and I wanted to keep the interface inline

Reply:
But this is just adding complexity to a call. And you can receive a different(bigger) amount than the value

Reply:
That's not an issue, too bad for the person who sent more. Which one would you prefer, to be different than the other connector interfaces or this redundant parameter?
I'm fine with either but leaning towards the second one.

Reply:
I tend to agree with @kstdl here. I was trying to use it and it feels weird.